### ISSUE TITLE
refactor service configuration from Salt (at provisioning) to systemd (at boot)

### ISSUE THREAD

**Issue by cfm** [freedomofpress/securedrop-workstation#1004]
* [x] I have searched for duplicates or related issues

## Description

@zenmonkeykstop asked this morning whether #1001 is sufficient for *all* VM-level configuration, not just keys and values.  I think we'll still want to use systemd units with [`ConditionHost`](https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html#ConditionHost=) conditions to enable individual services based on the hostname configured by Salt (and enforced by dom0 tests).

## How will this impact [SecureDrop/SecureDrop Workstation users](https://github.com/freedomofpress/securedrop-ux/wiki/Users)?

No user implications.

## How would this affect the SecureDrop Workstation [threat model](https://github.com/freedomofpress/securedrop-workstation/#threat-model)?

Along with #1001, this assumes we are comfortable with runtime (boot-time) configuration of VMs' roles and services, except for secrets.

**Comment by cfm** [comment]
Switching to Qubes services makes sense, @deeplow.  Arguably it extends #1001's configuration injection to enabling services by analogy, which I like.


<https://github.com/freedomofpress/securedrop-workstation/issues/1004#issuecomment-2095970555>:

> Disadvantages:
> 
> * **salt being salt** - salt is good at setting state, but not removing it. If we ever needed to disable a qubes service, we'd need to explicitly remove it in the salt configuration

I guess I take this for granted for as long as we're using Salt in this way at all.  :-)

**Comment by legoktm** [comment]
I think I lean in the direction of using ConditionHost, because I think it better fits our goal of keeping in-VM stuff in packages and using salt for dom0 things.

As a practical case is if we want to add a new "sd-log-whatever" service in a VM, we'd have to also do a corresponding workstation patch to enable the Qubes service through dom0, and gate the client release on the workstation one.z

**Comment by legoktm** [comment]
Re: `immune to qube rename side-effects (if they ever happen)` - one consideration could be to set a `sd-app` Qubes service on the sd-app VM, `sd-log` service on sd-log VM, etc. and then all of the services in VM can use `ConditionPathExists=/var/run/qubes-service/<VM name>`, which does feel a little less brittle than the hostname, but also allows for a 1:1 mapping between systemd service and VM. I'm not sure that extra level indirection is worth it though.

**Comment by legoktm** [comment]
> But other ideas may come up in the meeting we're having later.

Marek's point about wanting to set it in multiple VMs was pretty convincing to me. In theory we could do something like:
```
ConditionHost|=sd-app
ConditionHost|=sd-log
```

But I think that is less clean than the one ConditionPathExists. So, I'm down to move forwards with Qubes services, and if we end up running into problems, we can always revisit/adjust course.

**Comment by legoktm** [comment]
> Make disposable + provision via systemd + qubes services:
> * sd-proxy

Once https://github.com/freedomofpress/securedrop-workstation/pull/1035 lands, proxy is fully ready to be disposable! (I'm not sure why it has the mime handling enabled, nothing in that VM should be opening other files...)

**Comment by zenmonkeykstop** [comment]
Wasn't there mime handling config added in sd-proxy specifically to avoid it opening files?

**Comment by rocodes** [comment]
> Wasn't there mime handling config added in sd-proxy specifically to avoid it opening files?

Yes:  https://github.com/freedomofpress/securedrop-workstation/blob/d72c73ceb90db81bc8c93ee6c8312e4e9a3f9122/sd-proxy/mimeapps.list#L4 

sidebar: istr Marek mentioning a better way to deny this kind of functionality rather than trying to compete with all the places that mime handling could be introduced, and rather than having to specify every filetype, which has been a source of errors for  us in the past.  `qubes-core-agent-linux` does contain some mimetype overriding: it looks like they ship both `mime-override` and `xdg-override` (look in `/usr/share/qubes/` for respective directories) so it would be cool if we could do something similar.

But in any case for the purposes of this PR, I think we could either use the systemd approach that we're planning for other VMs, or just create the symbolic link to the "default" mime handling (which I think is just used for the proxy?) in the deb postinst and then override it in the other vms.

**Comment by cfm** [comment]
#1051, specifically ebcabf6fcc25a33d22ef7b8a1cf1d7960e5732c3, demonstrates using a Qubes service as the trigger for a `ConditionPathExists`-based systemd service that applies further configuration from QubesDB.
